tests/microgrid/test_client.py

```diff
@@ -130,7 +130,7 @@ async def test_components(self) -> None:
             }
 
         finally:
-            await server.stop(grace=1.0)
+            assert await server.graceful_shutdown()
```

Comment:
This is a tricky one IMHO, just because a mock server doesn't want to shut down, it doesn't really mean that the test failed. The other option is also not great, so for me it can go either way, but I wanted to put it out there just for general consideration.

Reply:
Yes I know :( 
But it is the only way. Otherwise it will crash in random test :/

Reply:
And I'd think having clean test outputs is a lot more valuable than being flexible towards mock server behaviour.

Reply:
Yeah, the ideal would be to have an "error" result to know the test didn't fail but there was something about the test itself that went wrong. But I agree is better to fail the test than to succeed it.